### DISCUSSION TITLE
Using ISettingProviver in Application.Tests project

### DISCUSSION THREAD

**Discussion by mc3000** [abpframework/abp#8078] [Q&A]
I am using ABP Framework with a gremlin graph database.

I use ISettingsProvider to read the database settings from my appsettings.json inside my Repository, via DI.

 protected readonly ISettingProvider _settingProvider;

        public GremlinBaseRepository(ISettingProvider settingProvider)
        {
            _settingProvider = settingProvider;
        }

I am then able to read the settings...

string host = (await _settingProvider.GetOrNullAsync("GremlinDbHost"));

I was hoping to create some integration tests inside the application.tests project and override the db settings with a different database. Is there a way to set up an ISettingsProvider  inside a test module that would override the settings in the repository?

**Comment by maliming** [comment]
You can try `Custom Setting Value Providers` in the unit test.

https://docs.abp.io/en/abp/latest/Settings#custom-setting-value-providers

**Reply by mc3000** [reply]
Thanks for your reply, do you know how I can set up a SettingProvider in my test?

When I new up my repository inside the test I need to pass in a SettingProvider...

var repo = new GremlinRepository(settingProvider);

In asp boilerplate I could do

var _settingManager = LocalIocManager.IocContainer.Resolve<ISettingManager>();

Is there an equivalent in abp framework?
